### PR TITLE
enhance: Adds support to collect the json output of ceph commands

### PR DESCRIPTION
**Pull request by ashishranjan738**
This commit adds support to collect the json output of ceph commands that can be used by json parsers to validate the results against validation rules.

Signed-off-by: Ashish Ranjan <aranjan@redhat.com>

### CODE REVIEW

File: must-gather/collection-scripts/gather_ceph_resources [hunk 1]
```diff
@@ -59,5 +57,12 @@ for ns in $(oc get cephcluster --all-namespaces --no-headers | awk '{print $1}')
             COMMAND_OUTPUT_FILE=${COMMAND_OUTPUT_DIR}/${ceph_commands[$i]// /_}
             oc -n ${ns} exec -it ${operator_pod_name} -- ${ceph_commands[$i]} >> ${COMMAND_OUTPUT_FILE}
         done
+
+        COMMAND_JSON_OUTPUT_DIR=${CEPH_COLLLECTION_PATH}/namespaces/${ns}/must_gather_commands/json_output
```
Comment (obnoxxx) [inline must-gather/collection-scripts/gather_ceph_resources]: It's mostly a taste/style thing, but wouldn't it be a little nicer to capture the two different output formats of the same list of commands in one loop instead of looping twice?

File: must-gather/collection-scripts/gather_ceph_resources [hunk 2]
```diff
@@ -30,10 +30,8 @@ ceph_commands+=("ceph mon dump")
 ceph_commands+=("ceph df")
 ceph_commands+=("ceph report")
 ceph_commands+=("ceph osd df tree")
-ceph_commands+=("ceph fs dump --format json-pretty")
```
Comment (obnoxxx) [inline must-gather/collection-scripts/gather_ceph_resources]: This command `ceph fs dump` and also `ceph health detail` below are removed completely from the list of commands. Are we not losing any piece of information here? (I.e. is the same information in the json output of other commands?)

File: Makefile [hunk 3]
```diff
@@ -3,12 +3,12 @@ all: clean ocs-operator ocs-must-gather
 .PHONY: ocs-operator
 ocs-operator:
 	@echo "Building ocs-operator image"
-	operator-sdk build quay.io/openshift/ocs-operator
+	operator-sdk build quay.io/ocs-dev/ocs-operator
```
Comment (obnoxxx) [inline Makefile]: This is a bit of nit-picking, but the changes that adapt the image names on quay.io are not related to the topic of the PR. I suggest moving them to a separate PR that clearly reflects this change.

File: Makefile [hunk 4]
```diff
@@ -3,12 +3,12 @@ all: clean ocs-operator ocs-must-gather
...
 
 .PHONY: ocs-must-gather
 ocs-must-gather:
 	@echo "Building ocs-must-gather image"
-	docker build -f must-gather/Dockerfile -t quay.io/openshift/ocs-must-gather:latest must-gather/
+	docker build -f must-gather/Dockerfile -t quay.io/ocs-dev/must-gather:latest must-gather/
```
Comment (obnoxxx) [inline Makefile]: Typo: it must be `quay.io/ocs-dev/ocs-must-gather:...`

Comment (obnoxxx) [reply Makefile]: (and also: separate PR)

File: must-gather/README.md [hunk 5]
```diff
@@ -6,7 +6,7 @@ that expands its capabilities to gather Openshift Container Storage 4 informatio
 
 ### Usage
 ```sh
-oc adm must-gather --image=ashishranjan738/ocs-must-gather
+oc adm must-gather --image=quay.io/ocs-dev/ocs-must-gather
```
Comment (obnoxxx) [inline must-gather/README.md]: This, too should go into a image-changing PR, imho.

File: must-gather/README.md [hunk 6]
```diff
@@ -17,7 +17,7 @@ You will get a dump of:
 - All namespaces (and their children objects) that belong to any ocs resources
 - All ocs CRD's definitions
 - All namspaces that contains ceph and noobaa
-- Output of the following ceph commands
+- Output of the following ceph commands in both json and standard format.
```
Comment (obnoxxx) [inline must-gather/README.md]: This hunk should go into the patch that changes the collection script to actually do the json collection.

File: Makefile (hunk 3 above)
Comment (kshlm) [reply Makefile]: I have submitter #20 to update the makefile with these, and few more changes. The Makefile changes can dropped from here.

**Comment by obnoxxx** [comment]
@ashishranjan738 you need to redo your patch on top of updated master now that #19 and #20 have been merged. The Makefile part has been taken care of. And please check the other requests/comments I put in above.

File: must-gather/collection-scripts/gather_ceph_resources (hunk 1 above)
Comment (ashishranjan738) [reply must-gather/collection-scripts/gather_ceph_resources]: yup done

File: must-gather/collection-scripts/gather_ceph_resources (hunk 2 above)
Comment (ashishranjan738) [reply must-gather/collection-scripts/gather_ceph_resources]: added `ceph fs dump` back and `ceph health detail` is already present

Comment (ashishranjan738) [reply must-gather/collection-scripts/gather_ceph_resources]: we are now collecting both json and standard output of the ceph commands.

File: Makefile (hunk 3 above)
Comment (ashishranjan738) [reply Makefile]: cool dropped.

File: Makefile (hunk 4 above)
Comment (ashishranjan738) [reply Makefile]: dropped the makefile as @kshlm has raised a PR for this. So marking this comment as resolved.

File: must-gather/README.md (hunk 5 above)
Comment (ashishranjan738) [reply must-gather/README.md]: A PR by @kshlm is raised for this so marking this as resolved.
